app/src/main/java/com/pajato/android/gamechat/common/InvitationManager.java

```diff
@@ -96,4 +246,97 @@ private boolean hasGroupInvite(@NonNull final Account account, @NonNull final St
         return true;
     }
 
+    /** Extend an invitation to join GameChat using AppInviteInvitation Intent */
+    public void extendAppInvitation(final FragmentActivity fragmentActivity) {
+        String firebaseUrl = FirebaseDatabase.getInstance().getReference().toString();
+        String dynamicLink = new Uri.Builder()
+                .scheme("https")
+                .authority(APP_CODE + ".app.goo.gl")
+                .path("/")
+                .appendQueryParameter("link", firebaseUrl)
+                .appendQueryParameter("apn", APP_PACKAGE_NAME)
+                .appendQueryParameter("afl", PLAY_STORE_LINK)
+                .appendQueryParameter("ifl", WEB_LINK)
+                .toString();
+
+        Log.i(TAG, "dynamicLink=" + dynamicLink);
+        Intent intent = new AppInviteInvitation.IntentBuilder(fragmentActivity.getString(R.string.InviteTitle))
+                .setMessage(fragmentActivity.getString(R.string.InviteMessage))
+                .setDeepLink(Uri.parse(dynamicLink))
+                .build();
+        fragmentActivity.startActivityForResult(intent, MainActivity.RC_INVITE);
+    }
+
+    /** Extend an invitation to join GameChat using AppInviteInvitation and specify a group to join. */
+    public void extendAppInvitation(final FragmentActivity fragmentActivity, final String groupKey) {
+        String firebaseUrl = FirebaseDatabase.getInstance().getReference().toString();
+
+        Log.i(TAG, "extendAppInvitation with groupKey=" + groupKey);
+        Group grp = GroupManager.instance.getGroupProfile(groupKey);
+        if (grp == null) {
+            Log.e(TAG, "Received invitation with groupKey: " + groupKey + " but GroupManager " +
+                    "can't find this group");
+            return;
+        }
+        firebaseUrl += "/groups/";
+        firebaseUrl += groupKey;
+
+        String dynamicLink = new Uri.Builder()
+                .scheme("https")
+                .authority(APP_CODE + ".app.goo.gl")
+                .path("/")
+                .appendQueryParameter("link", firebaseUrl)
+                .appendQueryParameter("apn", APP_PACKAGE_NAME)
+                .appendQueryParameter("afl", PLAY_STORE_LINK)
+                .appendQueryParameter("ifl", WEB_LINK)
+                .appendQueryParameter("commonRoomKey", grp.commonRoomKey)
+                .appendQueryParameter("groupName", grp.name)
+                .toString();
+
+        Log.i(TAG, "dynamicLink=" + dynamicLink);
+        Intent intent = new AppInviteInvitation.IntentBuilder(fragmentActivity.getString(R.string.InviteTitle))
+                .setMessage(fragmentActivity.getString(R.string.InviteMessage))
+                .setDeepLink(Uri.parse(dynamicLink))
+                .build();
+        fragmentActivity.startActivityForResult(intent, MainActivity.RC_INVITE);
+    }
+
+    public void onResult(@NonNull AppInviteInvitationResult result) {
```

Comment:
I have one word for this method and the author's style: Yuck.  This code looks very much like you lifted it pretty much intact from referenced docs.  I generally try to clean up code like this that I know will become core GC code so that maintainers, once they grok the code base, pretty much always face a similar style.  You, Bryan and I have similar styles with small differences.  External developers not so much.